test/e2e/managed_resource_vap_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2025 The KubeFleet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	testutils "go.goms.io/fleet/test/e2e/v1alpha1/utils"
+)
+
+const (
+	managedByLabel      = "fleet.azure.com/managed-by"
+	managedByLabelValue = "arm"
+	vapName             = "aks-fleet-managed-by-arm"
+	vapBindingName      = "aks-fleet-managed-by-arm"
+)
+
+// Helper functions for creating managed resources
+func createManagedNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				managedByLabel: managedByLabelValue,
+			},
+		},
+	}
+}
+
+func createUnmanagedNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+func expectDeniedByVAP(err error) {
+	var statusErr *k8sErrors.StatusError
+	Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Expected StatusError, got error %s of type %s", err, reflect.TypeOf(err)))
+	Expect(statusErr.ErrStatus.Code).To(Equal(int32(http.StatusUnprocessableEntity)), "Expected 422 UnprocessableEntity")
+	// ValidatingAdmissionPolicy denials typically contain these patterns
+	Expect(statusErr.ErrStatus.Message).To(SatisfyAny(
+		ContainSubstring("ValidatingAdmissionPolicy"),
+		ContainSubstring("denied the request"),
+		ContainSubstring("violates policy"),
+	), "Error should indicate policy violation")
+}
+
+var _ = Describe("ValidatingAdmissionPolicy for Managed Resources", Label("managedresource"), Ordered, func() {
+	BeforeAll(func() {
+		var vap admissionregistrationv1.ValidatingAdmissionPolicy
+		Expect(hubClient.Get(ctx, types.NamespacedName{Name: vapName}, &vap)).Should(Succeed(), "ValidatingAdmissionPolicy should be installed")
+
+		var vapBinding admissionregistrationv1.ValidatingAdmissionPolicyBinding
+		Expect(hubClient.Get(ctx, types.NamespacedName{Name: vapBindingName}, &vapBinding)).Should(Succeed(), "ValidatingAdmissionPolicyBinding should be installed")
```

Comment:
BeforeAll is mostly for setups. These two are testing implementations rather than expected behavior.